input/text/src/commonMain/kotlin/majestic/InputTextField.kt

```diff
@@ -1,117 +1,78 @@
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.LaunchedEffect
-import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
-import androidx.compose.runtime.getValue
 import androidx.compose.runtime.setValue
-import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.text.input.VisualTransformation
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
 import cinematic.watchAsState
 import kollections.firstOrNull
-import neat.Invalid
 import neat.Valid
 import neat.Validity
 import symphony.BaseField
 import symphony.toErrors
 
 @Composable
 fun InputTextField(
-    label: String,
-    value: String,
-    placeholder: String,
+    modifier: Modifier = Modifier,
+    field: BaseField<String>,
+    label: String = field.label.capitalizedWithAstrix(),
     keyboardOptions: KeyboardOptions = KeyboardOptions.Default,
     visualTransformation: VisualTransformation = VisualTransformation.None,
     trailingIcon: @Composable (() -> Unit)? = null,
     leadingIcon: @Composable (() -> Unit)? = null,
-    onValueChange: (String) -> Unit
-) {
-    val colorBlue = Color(red = 0x00, green = 0x61, blue = 0xFF)
-
-    Column(
-        modifier = Modifier
-            .fillMaxWidth()
-            .padding(vertical = 12.dp)
-    ) {
+    onChange: ((String) -> Unit)? = null
+) = InputTextField(
+    modifier = modifier,
+    field = field,
+    keyboardOptions = keyboardOptions,
+    visualTransformation = visualTransformation,
+    trailingIcon = trailingIcon,
+    leadingIcon = leadingIcon,
+    onChange = onChange,
+    label = {
         Text(
             modifier = Modifier.padding(bottom = 8.dp),
             text = label
         )
-        OutlinedTextField(
-            modifier = Modifier
-                .fillMaxWidth(),
-            singleLine = true,
-            value = value,
-            leadingIcon = leadingIcon,
-            trailingIcon = trailingIcon,
-            keyboardOptions = keyboardOptions,
-            visualTransformation = visualTransformation,
-            placeholder = {
-                Text(
-                    text = placeholder,
-                    color = Color.Black.copy(alpha = 0.4f),
-                    fontSize = 16.sp
-                )
-            },
-            colors = TextFieldDefaults.colors(
-                focusedContainerColor = Color.Transparent,
-                unfocusedContainerColor = Color.Transparent,
-                focusedIndicatorColor = colorBlue,
-                unfocusedIndicatorColor = Color.Black.copy(alpha = 0.2f),
-            ),
-            shape = RoundedCornerShape(8.dp),
-            onValueChange = { onValueChange(it) }
-        )
     }
-}
-
+)
 
 @Composable
 fun InputTextField(
+    modifier: Modifier = Modifier,
```

Comment:
Same as above, Modifier should be the first optional parameter in the parameter list; after all required parameters (except for trailing lambda parameters like card content) but before any other parameters with default values.